pkg/quantile/bin.go

```diff
@@ -0,0 +1,86 @@
+package quantile
+
+import (
+	"math"
+	"strings"
+)
+
+const (
+	maxBinWidth = math.MaxUint16
+)
+
+type bin struct {
+	k Key
+	n uint16
```

Comment:
The memory savings on the sketch-aggr side are worth it. 

Initial benchmarks showed the uint16 version performed roughly the same (extra computation is probably offset by less memory pressure).

Reply:
OK, I see. Also, how does this compare to alternative structures, such as skip lists or treemaps, especially with regard to insertion time? Is this method better because of the buffering and the bulk insertion logic?